docs/search.md

```diff
@@ -0,0 +1,139 @@
+# Search
+
+The Realm object has a `searchIndex` property, which in turn as a `search(...)` method. This is the method used to issue queries and get search results (card instance JSON-API representations) from the realm index.
+
+Note that search results are only as good as the most recent index, so if indexing is not working, or some cards are failing to be indexed, you may not get the results you expect. 
+
+## Search Queries
+
+The `SearchIndex#search(...)` method has one required argument, a search query. The simplest query is an empty object (`{}`), which returns all cards in the index.
+
+### Filters
+
+The Query object may have a `filter` property, which controls which cards are included in the result set.
+
+* `type`: allow specification of a particular card type. Also includes descendant cards in this card's adoption chain.
+* `eq`: allows specifying a card with a specific field value
+* `range`: allows specifying a card with a field value that is gt, gte, lt, or lte a specific value
+* `any`: allows an "OR" union of other filters
+* `every`: allows an "AND" union of other filters
+* `not`: allow negating another filter
+
+* `on`: may be used with `eq` or `range` to limit results to a particular card type.
```

Comment:
`on` is a lot more meaningful that what is being decribed here. `on` provides the card type context for a field. This is how we can disambiguage between a `Company.name` field and a `Country.name`. the `name` field field doesn't live in isolation. simply providing a predicate to filter by `name` field isnt enough. You need to say which card's `name` field you want to filter by--this is the function of `on`. It imbues the card definition context to all the filters, not just `eq` or `range` (here's an example using it with `not` https://github.com/cardstack/boxel/blob/f76dbedc9e828ff2b5ced0f57406e0fd4ae49b28/packages/host/tests/integration/search-index-test.ts#L3049-L3052).

Reply:
👍